tests/pyensign/test_publisher.py

```diff
@@ -0,0 +1,82 @@
+import pytest
+from asyncmock import call, patch
+
+from pyensign.publisher import Publisher
+from pyensign.exceptions import UnknownTopicError
+
+
+def async_iter(items):
+    async def next():
+        for item in items:
+            yield item
+
+    return next()
+
+
+class TestPublisher:
+    """
+    Tests for the high-level Publisher class.
+    """
+
+    @pytest.mark.parametrize(
+        "topic, kwargs, exception",
+        [
+            ("topic", {"client_id": "id"}, ValueError),
+            (None, {"client_id": "id", "client_secret": "secret"}, ValueError),
+            ("", {"client_id": "id", "client_secret": "secret"}, ValueError),
+        ],
+    )
+    def test_bad_params(self, topic, kwargs, exception):
+        with pytest.raises(exception):
+            Publisher(topic, **kwargs)
```

Comment:
What exactly is wrong with the params that should cause this to fail?

Reply:
The first case is that no client secret is provided, the second and third cases are that no topic is provided.